### PR TITLE
fix: Broken routing form public links(old links) on migrating the owner user to an organization

### LINKED ISSUE #13490: [CAL-3064] A user's routing forms link break after migration to an organization
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3064](https://linear.app/calcom/issue/CAL-3064/a-users-routing-forms-link-break-after-migration-to-an-organization)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #13490

Demo - https://www.loom.com/share/dfdd7a324cbd41ccb2030b5f45c481ae

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
See the loom for all the cases.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #13416** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13416?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13416?utm_source=stack-comment).

File: packages/prisma/zod-utils.ts [hunk 1]
```diff
@@ -329,6 +329,14 @@ export const userMetadata = z
       })
       .optional(),
     defaultBookerLayouts: bookerLayouts.optional(),
+    migratedToOrgFrom: z
```
Comment (hariombalhara) [inline packages/prisma/zod-utils.ts:332]: The data is already in there. Adding the schema to get type safety.

File: packages/prisma/zod-utils.ts [hunk 2]
```diff
@@ -329,6 +329,14 @@ export const userMetadata = z
...
+      .object({
+        username: z.string().or(z.null()).optional(),
+        lastMigrationTime: z.string().optional(),
+        reverted: z.boolean().optional(),
+        revertTime: z.string().optional(),
+      })
+      .optional(),
```
Comment (hariombalhara) [inline packages/prisma/zod-utils.ts:339]: Keeping it very loose, in case some of these props aren't set.

**Comment by CarinaWolli** [comment]
> How should this be tested?
See the loom for all the cases.

@hariombalhara What loom?

**Comment by hariombalhara** [comment]
lol here it is https://www.loom.com/share/dfdd7a324cbd41ccb2030b5f45c481ae

File: packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx [hunk 3]
```diff
@@ -97,11 +98,18 @@ function RoutingForm({ form, profile, ...restProps }: Props) {
       const allURLSearchParams = getUrlSearchParamsToForward(decidedActionWithFormResponse.response, fields);
       const decidedAction = decidedActionWithFormResponse.action;
 
+      form.migratedUserToOrgFrom;
```
Comment (CarinaWolli) [inline packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx:101]:
```suggestion
```

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
I am getting 404 now for the routing form links (`forms/[id]`). I didn't do anything related to orgs, this is just a normal team routing form.

Also routing-forms e2e are failing, which might be related to that

**Comment by hariombalhara** [comment]
Thanks @CarinaWolli 🙏 Fixed the issue. Waiting for tests to pass now.

**Review by CarinaWolli — DISMISSED** [review]
Works now and tests are passing. Fix looks good 👍

**Comment by hariombalhara** [comment]
Fixed conflicts with New Org Schema PR and custom route PR. Up for review again. Changed the logic a bit @CarinaWolli.

**Review by Udit-takkar — COMMENTED** [review]
LGTM. Tested and worked fine

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**